### ISSUE TITLE
incomplete initials

### ISSUE THREAD

**Issue by reschke** [ietf-tools/bibxml-service#264]
### Describe the issue

For instance, in https://bib.ietf.org/public/rfc/bibxml/reference.RFC.2119.xml : 

~~~
<author fullname="S. Bradner" initials="S" surname="Bradner"/>
~~~

But RFC 7991, Section 2.7.5 states:

> Some processors will post-process the value -- for instance, when it only contains a single letter (in which case they might add a trailing dot). Relying on this kind of post-processing can lead to results varying across formatters and thus ought to be avoided.

So the "initials" attribute should contain "S.".

### Code of Conduct

- [X] I agree to follow the [IETF's Code of Conduct](https://github.com/ietf-tools/.github/blob/main/CODE_OF_CONDUCT.md)

**Comment by ronaldtse** [comment]
The best answer to this question is this comment: https://github.com/ietf-tools/bibxml-service/issues/238#issuecomment-1192871487

1. The authoritative data from rfc-editor.org contains only the full name (i.e. "S. Bradner" here). The BibXML service is required to parse this single string into the `fullname` and `initials` attributes.
2. It is up to the XML RFC format on what goes into the `initials` attribute. Typically, it is the full name minus the surname. 

The RPC (as per @lbartholomew-rpc), requires the `initials` attribute to provide the "fully-formatted version of the initials of given names" as given by the RFC Editor.

i.e. 
* if the original was "S P", then `initials="S P"` (no dots)
* if the original was "S.P.", then `initials="S.P."` (with dots)
* if the original was "S P.", then `initials="S P."` (space then dot).

Does this work for you @reschke ?

**Comment by reschke** [comment]
This sounds right to me, but might break in tools which *do* try to process the initials.

**Comment by ronaldtse** [comment]
Thanks @reschke , I have the same feeling, yet it might be the case that the RPC’s needs are more important than other tools that try to process the initials (we don’t know which ones even do)?

Ping @rjsparks.

**Comment by rjsparks** [comment]
Your proposal above will work well for the RPC, and they would have trouble (would have to manually fix up) with what's happening at 2119.

**Comment by ronaldtse** [comment]
Thank you @rjsparks for the confirmation.

FYI @andrew2net @strogonoff.

**Comment by ronaldtse** [comment]
The fix in 
https://bib.ietf.org/public/rfc/bibxml/reference.RFC.2119.xml
```xml
<reference anchor="RFC2119" target="https://www.rfc-editor.org/info/rfc2119">
<front>
<title>
Key words for use in RFCs to Indicate Requirement Levels
</title>
<author fullname="S. Bradner" initials="S." surname="Bradner"/>
<date month="March" year="1997"/>
<abstract>
<t>
In many standards track documents several words are used to signify the requirements in the specification. These words are often capitalized. This document defines these words as they should be interpreted in IETF documents. This document specifies an Internet Best Current Practices for the Internet Community, and requests discussion and suggestions for improvements.
</t>
</abstract>
</front>
<seriesInfo name="BCP" value="14"/>
<seriesInfo name="RFC" value="2119"/>
<seriesInfo name="DOI" value="10.17487/RFC2119"/>
</reference>
```

This was fixed in #282 commit https://github.com/ietf-tools/bibxml-service/pull/282/commits/377d76f4ba26e6b675d8c4a19a04112cd756189b . Closing.
